### ISSUE TITLE
Partial deserialization.

### ISSUE THREAD

**Issue by sbeaudoin** [ExtendedXmlSerializer/home#393]
With v3.1.4, I receive the following XML from an external web source:

```
<?xml version="1.0" encoding="UTF-8"?>
<pin>
  <id type="integer">1309801580</id>
  <code>GBCT</code>
  <expires-at type="dateTime">2020-05-10T18:52:14Z</expires-at>
  <user-id type="integer">231221</user-id>
  <client-identifier>8c89f40c-e1f3-45e1-95bc-a45a047ce77f</client-identifier>
  <trusted type="boolean">false</trusted>
  <auth-token>z6ZM5jeSP19WroEQc2Xy</auth-token>
  <auth_token>z6ZM5jeSP19WroEQc2Xy</auth_token>
</pin>
```
the provided class for it is:
```
[XmlRoot("pin")]
public sealed class Pin
{
        [XmlElement("id")]
        public int ID { get; set; }

        [XmlElement("code")]
        public string Code { get; set; }

        [XmlElement("expires-at")]
        public DateTime ExpiresAtUTC { get; set; }

        [XmlElement("auth-token")]
        public string AuthToken { get; set; }

        [XmlElement("client-identifier")]
        public Guid ClientIdentifier { get; set; }
}
```
I instantiate the serializer and deserialize with the following.  This is a piece of the method who except a generic TReturn type.  In this case, a Pin class.

```
string xml = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);

IExtendedXmlSerializer serializer = new ConfigurationContainer().EnableImplicitTypingFromNamespace<Pin>().Create();
using (StringReader stringReader = new StringReader(xml))
{
        returnObj = (TReturn)serializer.Deserialize<TReturn>(stringReader);
}
```
Every property of the resulting class is populated correctly, except AuthToken who is always empty.  Is it a bug, or something I do wrong?

**Comment by sbeaudoin** [comment]
It works correctly if I add this to the Pin class:
```
[XmlElement("auth_token")]
public string AuthToken2 { get; set; }

[XmlElement("user-id")]
public string UserID { get; set; }

[XmlElement("trusted")]
public bool Trusted { get; set; }
```
If I remove the UserID and Trusted properties, it still don't work.  Does it means I must absolutely have a property for each element of the xml?  If so, it would be a good idea to either be able to receive an exception, telling me it can't complete, or else, have a mode where I can deserialize a partial class.

**Comment by sbeaudoin** [comment]
Last note.  If I remove the AuthToken2 property, it still works.

**Comment by Mike-E-wins** [comment]
Hi @sbeaudoin thanks for writing in. I took some time to look at this and you are correct the serializer is having problems with unknown content.  We probably show throw by default but that would be a breaking change.  In the meantime, we do have control over this, via the `WithUnknownContent` configuration:

```
var serializer = new ConfigurationContainer().EnableImplicitTyping(typeof(Pin))
			                                 .WithUnknownContent()
			                                 .Throw()
			                                 .Create();
```

Using your provided XML with the above configured container I get the following exception:

```
System.Xml.XmlException : Unknown/invalid member encountered: 'user-id'. Line 6, position 4.
```

ExtendedXmlSerializer is best used when XML is created and loaded from the serializer.  For scenarios where the XML is not created by the serializer, the recommendation is to load and transform it via XSLT before passing it to the serializer.  More details are found here:

https://github.com/ExtendedXmlSerializer/home/wiki/FAQs#systemxmlserializer-vs-extendedxmlserializer

Please let me know if you have any further questions and I will do my best to assist you. 👍

**Comment by sbeaudoin** [comment]
Thank you for the fast answer.  I sincerely tried to find this "switch" myself for a few hours.  Even now, if I search WithUnknownContent in the API reference, nothing come in the result.  I will try this library a little longer, in evaluation mode this time, hoping the documentation will be updated soon.  The main reason I use it is for the support of Enums, particularly,  I am hoping it support Flags Enums.

**Comment by Mike-E-wins** [comment]
Yeah it actually gets better, @sbeaudoin.  `WithUnknownContent().Continue()` (not to be confused with ABORT which is what happens without any configuration) does exactly what you are looking for:

https://github.com/ExtendedXmlSerializer/home/blob/35b82c89667fef7f46356b4df247ce19a3691f4e/test/ExtendedXmlSerializer.Tests.ReportedIssues/Issue393Tests.cs#L12-L40

As for the documentation we made a big push for 3.0 but there are some rough spots.  I was quite enamored with DocFX on first sight, but after using it along with its "search" as you have pointed out it leaves much to be desired.  I have tagged this with the `documentation` tag so it will show up for those that search for articles with this tag, but I think a section in the wiki explaining this would be valuable.
